### ISSUE TITLE
Internationalization. How to update references to keys (.tr)

### ISSUE THREAD

**Issue by zjjt** [jonataslaw/getx#1963]
### Discussed in https://github.com/jonataslaw/getx/discussions/1962

<div type='discussions-op-text'>

<sup>Originally posted by **zjjt** November  5, 2021</sup>
Hi @jonataslaw  ive been using getx for professional projects for a while and first let me congratulate you and the collaborators on this project for such an awesome framework.

As of my last project , i have been using a class to hold references by way of static variables to my translations(english and french) in the form of
![Screen Shot 2021-11-05 at 23 06 41](https://user-images.githubusercontent.com/15849390/140587933-c6ed1b28-9261-4db4-815a-353d4395a56f.png)

I have also created links to switch langages on the fly so that upon clicking them, the static variables are refreshed as shown below:
![Screen Shot 2021-11-05 at 23 11 42](https://user-images.githubusercontent.com/15849390/140588108-3dec0a43-d8d4-44aa-850b-fc330cf5149e.png)

Unfortunately i noticed that some of my references do not get updated .In fact i also use some of those strings in controller's methods , some of them are built in the onInit method 

I'd like to know if there is a way to get all my references to my strings updated as soon as they are reinitialized.

Regards.
</div>

**Comment by ivofernandes** [comment]
To update the translations run     

```
Get.clearTranslations();
Get.addTranslations(keys);
```

**Comment by zjjt** [comment]
HI @ivofernandes sorry for the late reply...Unfortunately it doesnt fix my issue.

It does switch the languages on the fly, but my issue is with widgets that are built using a method such as generating a list of widget to populate a Column for example. Those do not switch the language they are built with.

Do you have any other suggestion i may apply?

**Comment by ivofernandes** [comment]
On android studio you can drilldown in the .tr and check if they are being called or not, because you want it to be called after switching the language

In your screenshot of the issue you have static variables, may that be the issue? As the variable is static the tr will only be called once, and more, as it's static will stay in memory forever while using the app

**Comment by zjjt** [comment]
@ivofernandes You are right i think it might be the issue. I do recreate those static variables by using the **onInit** method shown in the capture so that when the language is switched the variables are reassigned. It works for some but for widgets that are build inside a controller via a method like the following example they wont refresh : 
`Widget buildExample(){
  return Text(Strings.sampleText);//the sampleText variable is declared like this static String sampleText="sampleText".tr; 
}
@override
void onInit(){
var textWidget=buildExample();
}
`

**Comment by zjjt** [comment]
@ivofernandes I'll try removing the static keyword and let you know here if it fixed the issue.
